Neos.Ui/core/src/application/LinkTypes/Web/Web.tsx

```diff
@@ -75,40 +82,39 @@ export const Web = makeLinkType<WebLinkModel>('Sitegeist.Archaeopteryx:Web', ({c
                 <div style={{ display: 'grid', gridTemplateColumns: '160px 1fr', minWidth: '600px' }}>
                     <Field<string>
                         name="protocol"
+                        format={value => {
+                            if(value === undefined){
+                                form.change(`${prefix}.protocol`, protocol);
+                            }
+                            return value;
+                        }}
                         initialValue={model?.protocol ?? 'https'}
                         validate={value => {
                             if (!value) {
                                 return i18n('Sitegeist.Archaeopteryx:LinkTypes.Web:protocol.validation.required');
                             }
                         }}
                     >{({input}) => (
-                        <SelectBox
-                            onValueChange={input.onChange}
-                            allowEmpty={false}
-                            value={input.value}
-                            options={[{
-                                value: 'https',
-                                label: 'HTTPS',
-                                icon: 'lock'
-                            }, {
-                                value: 'http',
-                                label: 'HTTP',
-                                icon: 'unlock'
-                            }]}
-                        />
+                        <>
+                            <SelectBox
+                                onValueChange={input.onChange}
+                                allowEmpty={false}
+                                value={input.value}
+                                options={[{
+                                    value: 'https',
+                                    label: 'HTTPS',
+                                    icon: 'lock'
+                                }, {
+                                    value: 'http',
+                                    label: 'HTTP',
+                                    icon: 'unlock'
+                                }]}
+                            />
+                            {input.value !== undefined && setProtocol(input.value)}
```

Comment:
This is not the right place to change the State. Please addjust the State in the onValeChange function. This kind of code is also unclear and miss leading. Please do not use this code style.

Reply:
I try to do that in onValeChange scope but, I could not get value there.